pkg/validate/container.go

```diff
@@ -109,6 +110,39 @@ var _ = framework.KubeDescribe("Container", func() {
 			testExecSync(rc, containerID)
 		})
 	})
+
+	Context("test container with volume", func() {
+		var podID string
+		var podConfig *runtimeapi.PodSandboxConfig
+
+		BeforeEach(func() {
+			podID, podConfig = createPodSandboxForContainer(rc)
+			pullImageList(ic, []string{defaultContainerImage})
+		})
+
+		AfterEach(func() {
+			By("stop PodSandbox")
+			rc.StopPodSandbox(podID)
+			By("delete PodSandbox")
+			rc.RemovePodSandbox(podID)
+		})
+
+		It("test start container with volume", func() {
+			By("create host path and flag file")
+			hostPath := "/test" + podID
+			flagFile := "test.file"
+			err := os.Mkdir(hostPath, 0777)
+			framework.ExpectNoError(err, "failed to create host path %s: %v", hostPath, err)
+			_, err = os.Create(hostPath + "/" + flagFile)
+			framework.ExpectNoError(err, "failed to create volume file %s: %v", flagFile, err)
+
+			By("create container with volume")
+			containerID := createVolumeContainer(rc, "container-with-volume-test-", podID, podConfig, hostPath, flagFile)
+
+			By("test start container with volume")
+			testStartContainer(rc, containerID)
```

Comment:
You may want to make sure the container consistently running.
Actually, it would be easier, if you just run `test -f $PATH_IN_THE_CONTAINER`, and check the exit status.